src/bin/vip-logs.js

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env node
+// @flow
+/**
+ * Internal dependencies
+ */
+import command from 'lib/cli/command';
+import { rollbar } from 'lib/rollbar';
+import { trackEvent } from 'lib/tracker';
+import * as logsLib from 'lib/logs/logs';
+import * as exit from 'lib/cli/exit';
+
+const LIMIT_MAX = 5000;
+const LIMIT_MIN = 1;
+const ALLOWED_TYPES = [ 'app', 'batch' ];
+
+export async function getLogs( arg: string[], opt ): Promise<void> {
+	validateInputs( opt.type, opt.limit, opt.env );
+
+	const trackingParams = {
+		command: 'vip logs',
+		org_id: opt.app.organization.id,
+		app_id: opt.app.id,
+		env_id: opt.env.id,
+		type: opt.type,
+		limit: opt.limit,
+	};
+
+	await trackEvent( 'logs_command_execute', trackingParams );
+
+	try {
+		const logs = await logsLib.getRecentLogs( opt.app.id, opt.env.id, opt.type, opt.limit );
+
+		for ( const { timestamp, message } of logs ) {
+			console.log( `${ timestamp } ${ message }` );
+		}
+	} catch ( error ) {
+		rollbar.error( error );
+
+		await trackEvent( 'logs_command_error', { ...trackingParams, error: error.message } );
+
+		return exit.withError( error.message );
+	}
+
+	await trackEvent( 'logs_command_success', trackingParams );
+}
+
+export function validateInputs( type: string, limit: number, env: Object ): void {
+	if ( ! env.isK8sResident ) {
```

Comment:
This isn't an input and seems like it should just be validated on the Parker side. This would eliminate the need for to query twice

Reply:
That makes sense 👍 

We'll need to update the error message on the GOOP side since it references k8s.

Reply:
@chriszarate I considered the provided `@app.environment` as an input for the logs command (maybe it causes some confusion... any suggestions?) because I used the common `command` function:
https://github.com/Automattic/vip/blob/a947ccd9b00630d487cd384700b2baa579a507ac/src/bin/vip-logs.js#L77-L82
This function queries the app and environment before starting the `vip logs` logic.

I totally agree that we should have this check on the API side. This was just a check to avoid another unnecessary query to the API.